### PR TITLE
feat: support invisible columns

### LINKED ISSUE #14367: Feature Request: Support invisible columns
### Feature Description

MySQL supports invisible columns that are valid to use explicitly, but that won't be included when using `*` expressions.

Vitess does not support this, and it should.

```sql
mysql> CREATE TABLE t1 (
    ->   i INT,
    ->   j DATE INVISIBLE
    -> );
Query OK, 0 rows affected (0.02 sec)

mysql> insert into t1 (i,j) values (0, '2023-01-01');
Query OK, 1 row affected (0.00 sec)

mysql> select * from t1;
+------+
| i    |
+------+
|    0 |
+------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by systay**
## Description
Adds supports for invisible columns. An invisible column is normally hidden to queries, but can be accessed if explicitly referenced.

As an illustration of when invisible columns may be useful, suppose that an application uses `SELECT *` queries to access a table, and must continue to work without modification even if the table is altered to add a new column that the application does not expect to be there. In a SELECT * query, the * evaluates to all table columns, except those that are invisible, so the solution is to add the new column as an invisible column. The column remains “hidden” from SELECT * queries, and the application continues to work as previously. A newer version of the application can refer to the invisible column if necessary by explicitly referencing it.

## Related Issue(s)
Fixes #14367

## Checklist
-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by harshit-gangal** [comment]
backport question on issue: https://github.com/vitessio/vitess/issues/14367#issuecomment-1780628175

File: proto/vschema.proto [hunk 1]
```diff
@@ -126,6 +126,7 @@ message AutoIncrement {
 message Column {
   string name = 1;
   query.Type type = 2;
+  bool invisible = 3;
```
Comment (harshit-gangal) [inline proto/vschema.proto:129]: I do not think we need this information in proto. possibly we directly use the proto struct so there might be not a better alternative here.

Comment (harshit-gangal) [reply proto/vschema.proto:129]: I checked, looks like It does not need to be part of proto.

Comment (dbussink) [reply proto/vschema.proto:129]: @harshit-gangal I wonder though, do we need it in the proto for older versions? Or could we also fix it there without that?

Comment (systay) [reply proto/vschema.proto:129]: I think that we should be consistent and allow this setting in the proto until we deprecate the manual specifying of column information in the vschema.

Comment (harshit-gangal) [reply proto/vschema.proto:129]: currently due to authoritative column information provided by vschema proto, we might keep this information and use it if provided. so let's keep this field till we do not deprecate this in vschema.

**Review by harshit-gangal — APPROVED** [review]

**Review by dbussink — APPROVED** [review]

**Pull request merged**